Startup tunes

saidinesh5 opened the pull request:
Initial support for Startup tunes in Bluejay ESC

Requires: https://github.com/mathiasvr/bluejay/pull/8

saidinesh5 committed:
feat: Initial support for Bluejay Startup Melody

mathiasvr committed:
Only display melody settings for supported layouts

mathiasvr committed:
Remove damping setting

mathiasvr committed:
Improve frequency conversion accuracy

mathiasvr commented:
Hi, I added a few commits.

I have now measured the beep pulses and made the frequency conversion more accurate, especially for higher frequencies.
I think this somehow messed up the timing of your melody a bit? However, this is now consistent with how it's played on the ESC. Not sure if this is related to your `todo` about bpm jumping around?

saidinesh5 committed:
ESC tab: Disable Accept button if there are no changes to the melody

saidinesh5 committed:
rtttl.js: Improve the rtttl <-> bluejay startup melody conversion

saidinesh5 committed:
bluejay_eeprom_layout: Minor fix to serializing startup melody

saidinesh5 committed:
rtttl.js: Add error reporting to Rtttl.toBluejayStartupMelody()

saidinesh5 committed:
Store Rtttl metadata in the first 4 bytes of Bluejay Startup Melody

This should make editing bluejay startup tunes a lot less jumpy.
The bluejay startup melody format is now:
[
 [2 bytes of bpm],
 [1 byte of default octave],
 [1 byte of default duration],
 [62 (Temp4, Temp3) values]
]

saidinesh5 committed:
bluejay_defaults: Add defaults for layout version 203

mathiasvr committed:
Improve melody playback timing and decrease volume

mathiasvr commented:
@saidinesh5 I tried to fix the playback by not using imprecise `setTimeout`.
Can you test if this works for you too [startup_tunes_playback](https://github.com/mathiasvr/bluejay-configurator/tree/startup_tunes_playback).

saidinesh5 committed:
Clean up MelodyComponent

saidinesh5 committed:
Bluejay startup melody: Add support for "Play All"

saidinesh5 committed:
Melody Editor: Log error messages to GUI.log

saidinesh5 committed:
Show a generic error message when Rtttl parsing fails

saidinesh5 committed:
Add a showMelodyEditor checkbox

mathiasvr committed:
Vertically align checkbox

mathiasvr committed:
Use this instead of self

mathiasvr committed:
fix some state bugs

mathiasvr committed:
Hide play all

saidinesh5 committed:
Fix playall bugs

AudioContext must be closed once done playing a tune
Also fixed some state handling errors when stopping a tune

mathiasvr committed:
Keep melody editor displayed between reads

saidinesh5 committed:
MelodyComponent: Clear the internal state on accept

mathiasvr committed:
Update settings

mathiasvr committed:
Change melody button text

saidinesh5 committed:
Add support for clearing Settings components internal state

Using blheli_configurator's state: currentSettingsInstanceId
which gets incremented upon every read
